docs/source/_ext/ghfiles.py

```diff
@@ -0,0 +1,65 @@
+import os.path as path
+import subprocess
+import shlex
+from sphinx.util import logging
+from docutils import nodes
+logger = logging.getLogger(__name__)
+
+
+# use an old git trick, to get the top-level, could have used ../ etc.. but
+# this will be fine..
+top = subprocess.check_output(shlex.split(
+    "git rev-parse --show-toplevel")).strip().decode("utf-8")
+
+
+def make_ref(text):
+    """ Make hyperlink to Github """
+    full_path = path.join(top, text)
+    if path.isfile(full_path):
+        ref = "https://github.com/numba/numba/blob/master/" + text
+    elif path.isdir(full_path):
+        ref = "https://www.github.com/numba/numba/tree/master/" + text
+    else:
+        logger.warn("Failed to find file in repomap: " + text)
+        ref = "https://www.github.com/numba/numba"
+    return ref
+
+
+def intersperse(lst, item):
+    """ Insert item between each item in lst. """
+    result = [item] * (len(lst) * 2 - 1)
+    result[0::2] = lst
+    return result
+
+
+def ghfile_role(name, rawtext, text, lineno, inliner, options={}, content=[]):
+    """ Emit hyperlink nodes for a given file in repomap. """
+    my_nodes = []
+    if "{" in text:  # myfile.{c, h} - make two nodes
+        # could have used regexes, but this will be fine..
+        base = text[:text.find(".") + 1]
+        exts = text[text.find("{") + 1:text.find("}")].split(",")
+        for e in exts:
+            node = nodes.reference(rawtext, base + e, refuri=make_ref(base + e), **options)
+            my_nodes.append(node)
+    elif "*" in text:  # path/*_files.py - link to directory
+        # Could have used something from os.path, but this will be fine..
+        ref = text[:text.rfind("/") + 1]
```

Comment:
The comment acknowledges it, but you might as well swap this for the simpler to write and read:
```
ref = os.path.dirname(text) + "/"
```

Reply:
check